### PR TITLE
Avoid live-updating badgeCache object

### PR DESCRIPTION
**Pull request by maximbaz**
Addressing comments in https://github.com/browserpass/browserpass-extension/pull/173

### CODE REVIEW

**Comment by maximbaz** [comment]
Merging as I feel confident this is what you had in mind, if you have comments, do leave them 😉

File: src/background.js [hunk 1]
```diff
@@ -113,24 +114,24 @@ chrome.runtime.onInstalled.addListener(onExtensionInstalled);
...
+            let response = await hostAction(settings, "list");
             if (response.status != "ok") {
                 throw new Error(JSON.stringify(response));
             }
 
-            badgeCache.files = response.data.files;
+            badgeCache = {
+                files: response.data.files,
+                settings: settings,
+                expires: Date.now() + 60 * 1000,
```
Comment (erayd) [inline src/background.js]: Please don't use magic numbers.

File: src/background.js [hunk 2]
```diff
@@ -113,24 +114,24 @@ chrome.runtime.onInstalled.addListener(onExtensionInstalled);
...
+                isRefreshing: false
+            };
         }
 
-        if (badgeCache.files === null) {
-            return; // badge cache refresh in progress
+        if (badgeCache.isRefreshing) {
```
Comment (erayd) [inline src/background.js]: Doesn't this condition belong at the very beginning of the function?

Comment (maximbaz) [reply src/background.js]: You are right, it does, remnants of the previous logic 👍

File: src/background.js (hunk 1 above)
Comment (maximbaz) [reply src/background.js]: 👍  will extract

**Pull request merged**